header/verify.go

```diff
@@ -26,6 +26,11 @@ func (eh *ExtendedHeader) IsExpired() bool {
 	return !expirationTime.After(time.Now())
 }
 
+// IsRecent checks if header is recent against the given blockTime.
+func (eh *ExtendedHeader) IsRecent(blockTime time.Duration) bool {
+	return time.Now().Sub(eh.Time) <= blockTime // TODO @renaynay: should we allow for a 5-10 block drift here?
```

Comment:
What would speak against allowing for a drift?

Reply:
So the argument **for allowing a certain amount of drift** is that this method determines how often the `trustedHead` method would eagerly request an objective head from a trusted peer (in the case that pending is empty for some reason, not really likely though). It could essentially render our subjective head "useless" (depending on the block time of the network, which hasn't yet been definitively decided) if we have a very tight constraint that sbjHead cannot even be 1 block behind -- by "useless", I mean it wouldn't really even make sense to fetch sbjHead from store if we're very likely going to toss it and ask our trusted peers for their Head instead.

The argument **against drift** is that it could (likely not though because we still have pending cache that is always checked first) return a slightly outdated head for a service that is relying on a head that is as current as possible (the state service for example). If you take a look at #933, when we request a balance from our core node connection, it will use the Syncer's latest known tip of chain to determine at which height it will request the balance from. This means if the Syncer returns a header that is a few blocks in the past, the balance could be outdated. However, all of this is still subjective to the Syncer as the Syncer is the one service inside of node that is most up-to-date with what the current network head is.

Reply:
I hope this makes sense, and maybe @Wondertan can weigh in

Reply:
Thanks for the explanation, it helped me understand a bit better now. Sounds like a tradeoff that I can't judge properly yet - for example, not sure how negative it is to be asking the peers for their head more often. Feel free to close this!